justfile

```diff
@@ -0,0 +1,74 @@
+name := 'cosmic-comp'
+export APPID := 'com.system76.CosmicComp'
+
+rootdir := ''
+prefix := '/usr'
+
+base-dir := absolute_path(clean(rootdir / prefix))
+
+export SHARE_DIR := sharedir
+export BIN_DIR := base-dir / 'bin'
+export LIB_DIR := libdir
+
+bin-src := 'target' / 'release' / name
+bin-dst := base-dir / 'bin' / name
+
+libdir := base-dir / 'lib'
+sharedir := base-dir / 'share'
+
+# Default recipe which runs `just build-release`
+default: build-release
+
+# Runs `cargo clean`
+clean:
+    cargo clean
+
+# `cargo clean` and removes vendored dependencies
+clean-dist: clean
+    rm -rf .cargo vendor vendor.tar
+
+# Compiles with debug profile
+build-debug *args:
+    cargo build {{args}}
+
+# Compiles with release profile
+build-release *args: (build-debug '--release' args)
+
+# Compiles release profile with vendored dependencies
+build-vendored *args: vendor-extract (build-release '--frozen --offline' args)
+
+# Runs a clippy check
+check *args:
+    cargo clippy --all-features {{args}} -- -W clippy::pedantic
+
+# Runs a clippy check with JSON message format
+check-json: (check '--message-format=json')
+
+# Installs files
+install:
+    install -Dm0755 {{bin-src}} {{bin-dst}}
+
+install-bare-session:
```

Comment:
Should this maybe depend on `install`? The bare session without the binary isn't particular useful.

Reply:
Updated